### PR TITLE
Add Openstack overlay

### PR DESCRIPTION
**Pull request by HomayoonAlimohammadi**
### Overview
This PR adds overlay for Openstack. Loadbalancer relation is not yet finalized and it to be decided/implemented in the future.

### CODE REVIEW

File: overlays/openstack.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+applications:
+  easyrsa:
+    charm: easyrsa
+    channel: stable
+    num_units: 1
+    constraints: cpu-cores=1 mem=4G root-disk=16G
+    base: ubuntu@22.04
```
Comment (addyess) [inline overlays/openstack.yaml:7]: 🤮 

No, let's not.  I don't want to have easyrsa in our bundles.  This should be a certificate authority we try to not use anymore.

File: overlays/openstack.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    trust: true
+    base: ubuntu@22.04
+  openstack-cloud-controller:
+    charm: openstack-cloud-controller
+    base: ubuntu@22.04
+  cinder-csi:
+    charm: cinder-csi
+    base: ubuntu@22.04
+relations:
+  - [openstack-cloud-controller:certificates,            easyrsa:client]
```
Comment (addyess) [inline overlays/openstack.yaml:20]: kubecontrol v2 rids the need of this.

File: overlays/openstack.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  - [openstack-cloud-controller:kube-control,            k8s:kube-control]
+  - [openstack-cloud-controller:external-cloud-provider, k8s:external-cloud-provider]
+  - [openstack-cloud-controller:openstack,               openstack-integrator:clients]
+  - [easyrsa:client,                                     cinder-csi:certificates]
+  - [k8s:kube-control,              cinder-csi:kube-control]
```
Comment (addyess) [inline overlays/openstack.yaml:25]: dumb nit -- can you tidy up the spaces on this line to it matches the others?

File: overlays/openstack.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  - [openstack-integrator:clients,                       cinder-csi:openstack]
+  # TODO(Hue): Figure out what we should do with loadbalancer
+  # - [k8s:loadbalancer-external,     openstack-integrator:lb-consumers]
```
Comment (addyess) [inline overlays/openstack.yaml:28]: very much so... But for the time being we can deploy without it.

**Review by addyess — CHANGES_REQUESTED** [review]
Please rid us of easyrsa

**Comment by HomayoonAlimohammadi** [comment]
All done, thanks for the review @addyess!

File: overlays/openstack.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    charm: openstack-cloud-controller
+    base: ubuntu@22.04
+  cinder-csi:
+    charm: cinder-csi
+    base: ubuntu@22.04
+relations:
+  - [openstack-cloud-controller:kube-control,            k8s:kube-control]
+  - [openstack-cloud-controller:external-cloud-provider, k8s:external-cloud-provider]
+  - [openstack-cloud-controller:openstack,               openstack-integrator:clients]
+  - [easyrsa:client,                                     cinder-csi:certificates]
```
Comment (addyess) [inline overlays/openstack.yaml:17]: BURN it 🗡️ 

But i think cinder-csi will need to relate to kube-control

Comment (addyess) [reply overlays/openstack.yaml:17]: to use cinder-csi with kube-control, we'll need this merged
* https://github.com/canonical/cinder-csi-operator/pull/9

File: overlays/openstack.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    trust: true
+    base: ubuntu@22.04
+  openstack-cloud-controller:
+    charm: openstack-cloud-controller
+    base: ubuntu@22.04
+  cinder-csi:
+    charm: cinder-csi
+    base: ubuntu@22.04
+relations:
+  - [openstack-cloud-controller:kube-control,            k8s:kube-control]
```
Comment (addyess) [inline overlays/openstack.yaml:14]: This relation needs:
https://github.com/charmed-kubernetes/openstack-cloud-controller-operator/pull/4

File: overlays/openstack.yaml (hunk 5 above)
Comment (HomayoonAlimohammadi) [reply overlays/openstack.yaml]: Forgot this one, my bad.

**Comment by HomayoonAlimohammadi** [comment]
@addyess both PRs for kube control v2 and tls-certificates interface are merged, we should be good to go.
- https://github.com/canonical/cinder-csi-operator/pull/9
- https://github.com/charmed-kubernetes/openstack-cloud-controller-operator/pull/4

File: overlays/openstack.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  - [openstack-cloud-controller:external-cloud-provider, k8s:external-cloud-provider]
+  - [openstack-cloud-controller:openstack,               openstack-integrator:clients]
+  - [openstack-integrator:clients,                       cinder-csi:openstack]
```
Comment (addyess) [inline overlays/openstack.yaml:17]: ugh, sorry i know this is silly, but can we flip these?
```suggestion
  - [cinder-csi:openstack,                       openstack-integrator:clients]
```

File: overlays/openstack.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    trust: true
+    base: ubuntu@22.04
+  openstack-cloud-controller:
+    charm: openstack-cloud-controller
+    base: ubuntu@22.04
+  cinder-csi:
+    charm: cinder-csi
+    base: ubuntu@22.04
+relations:
+  - [openstack-cloud-controller:kube-control,            k8s:kube-control]
```
Comment (addyess) [inline overlays/openstack.yaml:14]: cinder-csi needs `kube-control` too

```suggestion
  - [openstack-cloud-controller:kube-control,            k8s:kube-control]
  - [cinder-csi:kube-control,                            k8s:kube-control]
```

File: overlays/openstack.yaml [hunk 9]
```diff
@@ -0,0 +1,17 @@
+applications:
+  openstack-integrator:
+    charm: openstack-integrator
+    num_units: 1
+    trust: true
+    base: ubuntu@22.04
+  openstack-cloud-controller:
+    charm: openstack-cloud-controller
+    base: ubuntu@22.04
```
Comment (addyess) [inline overlays/openstack.yaml:9]: so, this is a subordinate charm.  I don't think you can set the base b/c the series will be determined by the primary unit (the openstack-integrator)

File: overlays/openstack.yaml [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+  cinder-csi:
+    charm: cinder-csi
+    base: ubuntu@22.04
```
Comment (addyess) [inline overlays/openstack.yaml:12]: same about the base -- exclude here

**Review by addyess — CHANGES_REQUESTED** [review]
Alrighty, i'm so sorry - there are a few more issues here

**Review by addyess — APPROVED** [review]
LGTM

**Pull request merged**